managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -2642,10 +2642,10 @@ public void testGetNumberOfEntriesInStorage() throws Exception {
             managedLedger.addEntry(("entry-" + i).getBytes(Encoding));
         }
 
-        //trigger ledger rollover and wait for the new ledger created
-        Field stateUpdater = ManagedLedgerImpl.class.getDeclaredField("state");
-        stateUpdater.setAccessible(true);
-        stateUpdater.set(managedLedger, ManagedLedgerImpl.State.LedgerOpened);
+        // trigger ledger rollover and wait for the new ledger created
+        Awaitility.await().untilAsserted(() -> {
+           assertEquals("LedgerOpened", WhiteboxImpl.getInternalState(managedLedger, "state").toString());
```

Comment:
```suggestion
           assertEquals(managedLedger.getState(), State.LedgerOpened);
```

WhiteboxImpl.getInternalState is not a good way. When this variable name is changed, it is easy to miss this point.

I prefer to call the method, you can also add the `@VisibleForTesting` to the test method.

Reply:
> WhiteboxImpl.getInternalState is not a good way. When this variable name is changed, it is easy to miss this point.
> 
> I prefer to call the method, you can also add the `@VisibleForTesting` to the test method.

+100

Reply:
I agree. I guess this is a topic of another possible PR. I'll resolve this discussion so that we can merge the fix.